content/blog/2020-11-30-super-optimization/index.md

```diff
@@ -0,0 +1,49 @@
++++
+title = "Super Optimization"
+[extra]
+bio = """
+  Socrates Wong is a secound year PhD student at Cornell who intreset lies in Computer Archicture.
+"""
+latex = true
+[[extra.authors]]
+name = "Socrates Wong"
++++
+
+## Background
+Most complier optimization (peephole, dead code elimination, constant propagation, loop invariant code motion, common sub expression elimination) focus on transforming the program in a systematic manner, and the step taken by the optimizer are backed by formal rigorous proofs that are both sound and complete.  Superoptimizer is a new radical way of preforming optimization.  Instead of taking the traditional approach optimizers of transforming the input program, superoptimizer converts the problem in to a search space and attempts to find the shortest program that is able to provide the equivalent output.
```

Comment:
"formal rigorous proofs" is quite optimistic! Very few compilers are backed by formal proofs.